packages/core/src/types/IDIDManager.ts

```diff
@@ -124,6 +124,28 @@ export interface IDIDManagerGetOrCreateArgs {
   options?: object
 }
 
+export interface IDIDManagerUpdateArgs {
```

Comment:
The shape of this interface suggests that the update being done is mostly, if not only, to the locally known `IIdentifier` metadata.
If this is meant to update the DIDDocument, I think it should be more suggestive of that, by adding a parameter of type `DIDDocument` to represent the new DID doc.

I imagine something like this would be more suited for full document updates:
```ts
export interface IDIDManagerUpdateDocumentArgs {
  did: string,
  document: DIDDocument, // this could also be `Partial<DIDDocument>` // or a new type that can also link to local keys (read below)
  options?: Record<string, any> // these options get forwarded to the DID provider
}
```

---

If the new DID document contains new keys, there's also the question of whether the agent should be managing these keys or not.

If the agent already knows the keys, the agent could figure out which keys match the update and update the local `IIdentifier.keys` array accordingly.

Alternatively, the `verificationMethod` array from the `document` parameter could contain private key information, in which case these keys would also be imported by the `KeyManager`. In this later case, the `DIDDocument` data type would have to be replaced by an extended version that includes veramo specific key information. However, this would introduce some complexity, so it would be better if the user would first import the keys directly to the key manager and then call this update.

What do you think?

Reply:
The rationale is to always pass a raw DIDDoc along with key-pairs.

The workflow:

- Import keys, if needed for the specific DIDDoc (might already be managed). Handled by `didManagerImport` for an identifier or `keyManagerImport` for just keys.
- Update on ledger, handling validation logic ledger-side
- Return the identifier implementing `IIdentifier`

The provider is going to handle the ledger-specific args format and key-pairing in signature generation.
The `DIDManager`'s `didUpdateMethod` only takes care of passing the needed information to the provider handling the external update and updates the local identifier implementing `IIdentifier`.

Reply:
Also, about the DIDDoc, I'd prefer to pass it in the options, as they are being directly passed to the provider as well.

e.g. 
```typescript
export interface IDIDManagerUpdateArgs {
    did: string
    options?: {
        document: DIDDocument
    }
}
```

@mirceanis What do you think?